src/cpp/session/projects/SessionProjectContext.cpp

```diff
@@ -54,6 +55,34 @@ namespace projects {
 
 namespace {
 
+static std::unique_ptr<r_util::RPackageInfo> s_pIndexedPackageInfo = nullptr;
+
+void onDescriptionChanged()
+{
+   s_pIndexedPackageInfo.reset();
+
+   std::unique_ptr<r_util::RPackageInfo> pInfo(new r_util::RPackageInfo);
+   Error error = pInfo->read(projectContext().directory());
+   if (error)
```

Comment:
Nit: If this fails, we'll be left with no package info. Do you think it'd be better to wait to remove the old info until we've successfully read new info? (Or does no info more accurately reflect the state?)

Reply:
Good question. Perhaps it's okay to cache the 'invalid' state as well, since that could still be reused and give the correct answer for "is this a package project?". I'll make that change.